src/NUnitFramework/tests/Assertions/AdhocTestExecutionTests.cs

```diff
@@ -47,7 +47,7 @@ public void CanCallAssertWithoutTestExecutionContext(MethodInfo method)
                 throw testException;
         }
 
-#if !(NET35 || NET40 || NET45)
+#if !(NET35 || NET40 || NET45 || NET46) // NET46 should be removed when the framework project adds a net46 target
```

Comment:
We could consider adding a net46 target to the framework so that it can use AsyncLocal.

https://github.com/nunit/nunit/blob/30df2d13c0f67eeb6b3dc7f56eebd7c5cfb5c7de/src/NUnitFramework/framework/Internal/TestExecutionContext.cs#L144-L195

Reply:
Agreed. I've been looking for features in 4.6, 4.7 and 4.8 that would justify any of those targets but couldn't think of any. This might be one...